doc/patterns/tables.md

```diff
@@ -0,0 +1,282 @@
+
+Using Material React Table, we've created some abstractions that allow developers to easily implement tables using a standardised pattern, with minimal boilerplate and a UI consistent with the existing app.
+
+There are three main types of tables used throughout OMS, and we've created a different hook to implement each one:
+
+### Paginated tables
+
+These are tables that fetch a single **page** of data at a time, and therefore every change to the sorting or filtering requires an additional query to the server.
+
+The hook for this type of table is `usePaginatedMaterialTable`
+
+### Non-paginated tables
+
+These are tables that fetch the *full* data set in one query, and so all filtering/sorting is done in the front-end by the table library itself.
+
+The hook for this type of table is `useNonPaginatedMaterialTable`
+
+### "Simple" tables
+
+These are the tables used in the "Line Edit" views -- i.e. one line per batch, with inputs for stock allocation. They only have a handful of rows, so don't require any filtering/sorting UI and don't interact with the URL Query.
+
+The hook for this type of table is `useSimpleMaterialTable`
+
+These hooks are just wrappers around MRT's standard `useMaterialReactTable` hook, but with a lot of settings pre-defined with appropriate defaults for the given context. Note that *any* pre-defined value can still be overridden with explicit props.
+
+## Usage
+
+> [!TIP]
+> For a reference implementation example, please check out **Outbound Shipments**' `ListView`, `DetailView` and `OutboundLineEditTable` components for a version of each of these table types. This is the "OG" implementation of the new table library, so always refer to that if you're unsure how to do something, and branch out from those examples if new functionality is needed.
+
+All code related to these table implementations can be found in:
+
+```
+client/packages/common/src/ui/layout/tables/material-react-table
+```
+
+And all elements can be imported using the `@open-msupply/common` package.
+
+The basic implementation for a table is to use the hook to generate the `table` object (and `selectedRows` array), which is then passed to the `MaterialTable` component (and `Footer` if required):
+
+```tsx
+const { table, selectedRows } =
+    useNonPaginatedMaterialTable<StockOutLineFragment>({
+        tableId: 'outbound-shipment-detail-view',
+        columns,
+        data,
+        ...otherProperties
+        ),
+    });
+
+
+return <>
+        <MaterialTable table={table} />
+        // ...Other JSX
+        <Footer selectedRows={selectedRows} {...otherFooterProps}>
+    </>
+```
+
+> [!NOTE]
+> - Usage is the same for all three hooks
+> - The hook call is **typed** with a generic parameter, which should correspond to the type of the data **row**s
+
+### Input props
+
+**Required**:
+- `tableId`: Must be unique throughout the whole app — this is used as part of the data key when the table config is saved to local storage
```

Comment:
This makes me wonder - do you think we should make an enum or something for this? So it's easy to see which table IDs have already been used, and one click to the definition with a big comment saying each table must have unique id??